app/views/core/VueComponentView.js

```diff
@@ -0,0 +1,41 @@
+
+// This is the base backbone view to render any VueComponent.
+// Pass the vue component, propsData, base template into the constructor
+
+import RootView from 'views/core/RootView'
+import store from 'core/store'
+
+const silentStore = { commit: _.noop, dispatch: _.noop }
+
+module.exports = class VueComponentView extends RootView {
+  constructor (component, options) {
+    super(options)
+    const baseTemplate = options.baseTemplate || 'base-flat'  //base template, by default using base-flat
+    this.id = 'vue-component-view'
+    this.template = require('templates/'+ baseTemplate)
```

Comment:
Could we do an additional check prior to this ensuring that only valid base templates can be provided?

Reply:
Thats a good idea, will move it to a try-catch block to catch any error in importing the base template.